### PR TITLE
Answers are done 

### PR DESCRIPTION
**Pull request by VeiveHata**
the list of answers is displayed; upon confirmation the answer is deleted from list;
upon deleating answers all unapproved answers are deleted

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -1,10 +1,26 @@
-import { Component } from '@angular/core';
+import { Component, OnInit } from '@angular/core';
+import { AuthService } from './services/auth.service';
+import * as $ from 'jquery';
```
Comment (FrontendMentor) [inline src/app/app.component.ts]: Remove if you do not use it

File: src/app/components/answers/answers.component.ts [hunk 2]
```diff
@@ -1,21 +1,63 @@
...
         });
     }
+
+    ngOnInit() { }
+
+    getUserName(userId: string): string {
+        return  this.users.find((user: User) => user.key === userId).name;
+    }
+
+    sortQuestions(questions: Question[]): any {
```
Comment (FrontendMentor) [inline src/app/components/answers/answers.component.ts]: I think the return type is `Question[]`

File: src/app/components/answers/answers.component.ts [hunk 3]
```diff
@@ -1,21 +1,63 @@
...
+            if(question.answers) {
+                return this.inArray(question.answers) 
+                           .some((answer: Answer) => answer.status === false)
+            } else {
+              return false;
+            }
+        })
+    }
+
+    inArray(answers: any): any {
```
Comment (FrontendMentor) [inline src/app/components/answers/answers.component.ts]: try to replace type `any`, maybe there should be `inArray(answers: Answer[]): Answer[] | boolean {` 
or something similar

File: src/app/components/answers/answers.component.ts [hunk 4]
```diff
@@ -1,21 +1,63 @@
...
+
+    toApproveAnswer(questionKey: string, answerKey: string): void {
+        this.questionService.approveAnswer(questionKey, answerKey);
+    }
+
+    deleteUnapprovedAnswers(questionKey: string): void {
+        let allAnswers = this.questions.find((question: Question) => question.key === questionKey).answers;
+        let unproovedAnswers = Object.values(allAnswers)
+                               .filter((answer: any) => answer.status === false)
+                               .map((answer: any) => this.questionService.deleteUnapprovedAnswer(questionKey, answer.key));
```
Comment (FrontendMentor) [inline src/app/components/answers/answers.component.ts]: try to replace type `any`

File: src/app/components/questions/questions.component.ts [hunk 5]
```diff
@@ -26,17 +30,16 @@ export class QuestionsComponent implements OnInit {
...
-    }
 
-    setName(id: string): string {
-        let name: string;
-
-        this.users.forEach((user: User) => {
-            if (user.key === id) name = user.username;
-        });         
-        return name;
+        this.auth.authenticated.subscribe(authenticatedValue => {
```
Comment (FrontendMentor) [inline src/app/components/questions/questions.component.ts]: let's add a type for `authenticatedValue`

File: src/app/interfaces/answer.ts [hunk 6]
```diff
@@ -1,7 +1,8 @@
 export interface Answer {
 	answer: string,
-	date: Date,
-	questionId: string,
+	date: any,
```
Comment (FrontendMentor) [inline src/app/interfaces/answer.ts]: why did you change `Date` to `any`?

File: src/app/components/answers/answers.component.ts [hunk 7]
```diff
@@ -1,21 +1,68 @@
...
+        });       
+        return name;
+    }
+
+    sortQuestions(questions: Question[]): Question[] {
+        return questions.filter(question => {
+            if(question.answers) {
+                return this.inArray(question.answers) 
+                           .some((answer: Answer) => answer.status === false)
+            }
```
[the remark this answers is not in the document]
Comment (FrontendMentor) [reply src/app/components/answers/answers.component.ts]: @VeiveHata @adebos 
I think you can just use `return false` after condition

File: src/app/components/answers/answers.component.ts (hunk 4 above)
Comment (VeiveHata) [reply src/app/components/answers/answers.component.ts]: here match string. but.. it`s not string type. tried to set Answer[] to allAnswers. not correct to
